src/header/header.component.js

```diff
@@ -1,24 +1,46 @@
-import React, { useCallback, useState } from 'react';
+import React, { useContext } from 'react';
 import { Header, TopMenu } from './header.styled';
 import SearchForm from './searchForm/searchForm.component';
 import LogoComponent from '../global/components/logo/logo.component';
-import PosterForm from '../postersBlock/posterForm/posterForm.component';
+import { PosterForm } from '../postersBlock/posterForm/posterForm.component';
 import { ButtonMovie } from './addMovie/addMovie.styled';
+import { AppContext } from '../App';
+import Search_Button from '../../public/images/Search_Button.png';
+import { Main } from '../global/globalStyles';
+import { PosterInfoComponent } from './posterInfo/posterInfo.component';
+import { useToggle } from '../global/hooks/useToggle';
 
 const HeaderComponent = () => {
-  const [showAdd, setShowForm] = useState(false);
-  const changeStateModal = useCallback(() => {setShowForm(!showAdd);});
+  const { posterIdForHeader } = useContext(AppContext);
+  const [showAdd, setShowForm] = useToggle();
 
   return (
     <>
-      <Header>
-        <TopMenu>
-          <LogoComponent />
-          <ButtonMovie onClick={() => changeStateModal()} />
-          {showAdd  && <PosterForm showAdd={showAdd}  changeStateModal={() => changeStateModal()} />}
-        </TopMenu>
-        <SearchForm />
-      </Header>
+      {posterIdForHeader.selectedPoster === false ? (
```

Comment:
posterIdForHeader.selectedPoster === false
can be simplified !posterIdForHeader.selectedPoster